ui/ducks/send/send.js

```diff
@@ -0,0 +1,1325 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import abi from 'human-standard-token-abi';
+import log from 'loglevel';
+import contractMap from '@metamask/contract-metadata';
+import BigNumber from 'bignumber.js';
+import { addHexPrefix, toChecksumAddress } from 'ethereumjs-util';
+import { debounce } from 'lodash';
+import {
+  conversionGreaterThan,
+  multiplyCurrencies,
+  subtractCurrencies,
+} from '../../helpers/utils/conversion-util';
+import { GAS_LIMITS } from '../../../shared/constants/gas';
+import {
+  CONTRACT_ADDRESS_ERROR,
+  INSUFFICIENT_FUNDS_ERROR,
+  INSUFFICIENT_TOKENS_ERROR,
+  INVALID_RECIPIENT_ADDRESS_ERROR,
+  INVALID_RECIPIENT_ADDRESS_NOT_ETH_NETWORK_ERROR,
+  KNOWN_RECIPIENT_ADDRESS_ERROR,
+  MIN_GAS_LIMIT_HEX,
+  NEGATIVE_ETH_ERROR,
+} from '../../pages/send/send.constants';
+
+import {
+  addGasBuffer,
+  calcGasTotal,
+  generateTokenTransferData,
+  isBalanceSufficient,
+  isTokenBalanceSufficient,
+} from '../../pages/send/send.utils';
+import {
+  getCurrentChainId,
+  getGasPriceInHexWei,
+  getSelectedAddress,
+  getTargetAccount,
+} from '../../selectors';
+import {
+  displayWarning,
+  estimateGas,
+  hideLoadingIndication,
+  showConfTxPage,
+  updateTransaction,
+} from '../../store/actions';
+import {
+  fetchBasicGasEstimates,
+  setCustomGasLimit,
+  BASIC_ESTIMATE_STATES,
+} from '../gas/gas.duck';
+import {
+  SET_BASIC_GAS_ESTIMATE_DATA,
+  BASIC_GAS_ESTIMATE_STATUS,
+} from '../gas/gas-action-constants';
+import {
+  QR_CODE_DETECTED,
+  SELECTED_ACCOUNT_CHANGED,
+  ACCOUNT_CHANGED,
+} from '../../store/actionConstants';
+import { calcTokenAmount } from '../../helpers/utils/token-util';
+import {
+  checkExistingAddresses,
+  isDefaultMetaMaskChain,
+  isOriginContractAddress,
+  isValidDomainName,
+} from '../../helpers/utils/util';
+import { getTokens, getUnapprovedTxs } from '../metamask/metamask';
+import { resetResolution } from '../ens';
+import {
+  isBurnAddress,
+  isValidHexAddress,
+} from '../../../shared/modules/hexstring-utils';
+
+// typedefs
+/**
+ * @typedef {import('@reduxjs/toolkit').PayloadAction} PayloadAction
+ */
+
+const name = 'send';
+
+/**
+ * The Stages that the send slice can be in
+ * 1. UNINITIALIZED - The send state is idle, and hasn't yet fetched required
+ *  data for gasPrice and gasLimit estimations, etc.
+ * 1. ADD_RECIPIENT - The user is selecting which address to send an asset to
+ * 2. DRAFT - The send form is shown for a transaction yet to be submitted to
+ *  the network.
+ * 3. EDIT - The send form is shown for a transaction already submitted that
+ *  has not yet been signed (confirmed). This happens when the confirmation
```

Comment:
Only signed txs are sent to the network. I think DRAFT should be "not yet sent to the tx controller", and "EDIT" is "sent to the tx controller, but not yet signed or sent to network.

Reply:
That makes sense!

Reply:
I believe I captured this in the changes, let me know!